src/main/java/com/google/sps/servlets/GmailActionableEmailsServlet.java

```diff
@@ -0,0 +1,133 @@
+import com.google.sps.model.AuthenticatedHttpServlet;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.model.GmailClientImpl;
+import com.google.sps.utility.JsonUtility;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet to get actionable emails from user's gmail account, as defined by specific words in an
+ * email's subject line. Used by "Assign" panel on client.
+ */
+@WebServlet("/gmail-actionable-emails")
+public class GmailActionableEmailsServlet extends AuthenticatedHttpServlet {
+  private GmailClientFactory gmailClientFactory;
+
+  /** Create new servlet instance (called by java during HttpRequest handling */
+  public GmailActionableEmailsServlet() {
+    gmailClientFactory = new GmailClientImpl.Factory();
+  }
+
+  /**
+   * Create new servlet instance (used for testing)
+   *
+   * @param authenticationVerifier implementation of AuthenticationVerifier interface
+   * @param gmailClientFactory implementation of GmailClientFactory interface
+   */
+  public GmailActionableEmailsServlet(
+      AuthenticationVerifier authenticationVerifier, GmailClientFactory gmailClientFactory) {
+    super(authenticationVerifier);
+    this.gmailClientFactory = gmailClientFactory;
+  }
+
+  /**
+   * Retrieve the actionable emails from a user's gmail account, given a list of words/phrases to
+   * search for in the subject line.
+   *
+   * @param request HTTP request from client
+   * @param response Http response to be sent to client
+   * @param googleCredential valid, verified google credential object
+   * @throws IOException If a read/write issue arises while processing the request
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    GmailClient gmailClient = gmailClientFactory.getGmailClient(googleCredential);
+
+    List<String> subjectLinePhrases;
+    try {
+      subjectLinePhrases = getListFromQueryString(request, "subjectLinePhrases");
+    } catch (IllegalArgumentException e) {
+      response.sendError(400, "subjectLinePhrases must be present in request");
+      return;
+    }
+    if (subjectLinePhrases.isEmpty()) {
+      response.sendError(400, "subjectLinePhrases must be non-empty");
+      return;
+    }
+
+    boolean unreadOnly;
+    String unreadOnlyParameter = request.getParameter("unreadOnly");
+    if (unreadOnlyParameter == null) {
+      response.sendError(400, "unreadOnly must be present in request");
+      return;
+    }
+    if (!unreadOnlyParameter.equals("false") && !unreadOnlyParameter.equals("true")) {
+      response.sendError(400, "unreadOnly must be a boolean");
+    }
```

Comment:
Okay yes totally fair. I like the idea of the optional parameter. Thanks!

Also, to follow up here, this actually was tested using the methods we've used previously, but I think this method might actually be failing. The test checks if the error code is 400, which in this case it is. However, there is also a response, since the request continues processing erroneously. I'm going to file an issue (#128) to look at this in a separate PR (and ensure that only an error is returned, and nothing else, when an error is sent in a servlet)